src/main/kotlin/uk/gov/justice/digital/hmpps/pollpush/resource/QueueAdminResource.kt

```diff
@@ -0,0 +1,50 @@
+package uk.gov.justice.digital.hmpps.pollpush.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.pollpush.services.QueueAdminService
+
+@RestController
+@RequestMapping("/queue-admin", produces = [MediaType.APPLICATION_JSON_VALUE])
+class QueueAdminResource(private val queueAdminService: QueueAdminService) {
+
+  @PutMapping("/purge-dlq")
+  @PreAuthorize("hasRole('CASE_NOTE_QUEUE_ADMIN')")
```

Comment:
I wonder if we should also have a generic role `QUEUE_ADMINISTRATOR` they we use on all apps

Reply:
I wondered the same but couldn't come up with an argument that convinced me either way.